src/transaction.ts

```diff
@@ -969,6 +1029,21 @@ export class Snapshot extends EventEmitter {
     });
   }
 
+  /**
+   *
+   * @private
+   */
+  configureTagOptions(
+    singleUse?: boolean,
+    transactionTag?: string,
+    requestTag?: string
+  ): spannerClient.spanner.v1.IRequestOptions | null {
+    if (!transactionTag && !requestTag) {
+      return null;
+    }
+    return singleUse ? {requestTag} : {transactionTag, requestTag};
```

Comment:
I don't think this check is correct:
1. A `singleUse` transaction could in theory be a read/write transaction in the form of a single `Commit` call containing mutations.
2. A non-`singleUse` transaction is not by definition taggable, as it could also be a read-only transaction, which should not include transaction tags.

Reply:
As per documentation
> Single use transactions should not support setting a transaction tag.

In terms of read-only vs read-write transactions
Only `class Transaction` object constructor accepts `transactionRequestOptions` param and passes it to all its methods regardless whether `request tag` is set on per method basis.
`class Snaphot` (a read-only transaction) object does not accept any tags, this way only `request tag` could be set on per method basis.

Reply:
> Single use transactions should not support setting a transaction tag.

Hmm... I think the specs are a little ambiguous regarding this (although the above sentence is not ambiguous...). It also states the following:

> __Batch Methods__
Methods that batch Mutations together should support setting a transaction tag. They should
not support setting a request tag. This is because they are sent together as a single
CommitRequest.

The above can apply to both a multi-use and single-use read/write transaction. Single-use read/write transactions are not very well supported in the NodeJS client library (which is not necessarily a bad thing, as they are for advanced use only), but the following code snippet would execute a single-use read/write transaction:

```typescript
const database = instance.database('some-database');
const session = database.session();
await session.create();
const transaction = new Transaction(session);
transaction.upsert('Singers', {SingerId: 1, Name: 'Some Name'});
await transaction.commit();
```

The above transaction will in the current setup not include the transaction tag.


Regarding the transaction tag you're completely right that `Snapshot` does not accept `transactionRequestOptions`. Sorry for missing that one. That does indeed make it impossible to create a read-only transaction that would include a transaction tag on subsequent requests.

It is however possible to call `begin` on a `Snapshot` and include a transaction tag there. That transaction tag will be included in the `BeginTransaction` request, but it will not be used for subsequent requests on the read-only transaction. I think that will be confusing for users. I'm also not sure whether the backend would accept a `BeginTransaction` request for a read-only transaction that includes a transaction tag.